target/bin/listmailuser

```diff
@@ -90,15 +90,9 @@ function _alias_list_for_account
   local MAIL_ACCOUNT=${1}
 
   # `__db_list_already_contains_value` would be a more reliable check:
-  function _account_has_an_alias
-  {
-    local ANY_ALIAS='\S\+\s'
-    grep -qis "^${ANY_ALIAS}.*${MAIL_ACCOUNT}" "${DATABASE_VIRTUAL}"
-  }
-
-  if _account_has_an_alias
+  if grep -qis "\s${MAIL_ACCOUNT}$" "${DATABASE_VIRTUAL}"
   then
-    grep "${MAIL_ACCOUNT}" "${DATABASE_VIRTUAL}" | awk '{print $1;}' | sed ':a;N;$!ba;s/\n/, /g'
+    grep "\s${MAIL_ACCOUNT}$" "${DATABASE_VIRTUAL}" | awk '{print $1;}' | sed ':a;N;$!ba;s/\n/, /g'
```

Comment:
This should be all you need to add really, the `if` condition can stay as it was.

This still has that earlier problem of not working for an alias that is used for multiple accounts (_eg: `marketing@example.com` / `sales@example.com` aliased to several staff at a company_).

~~For the most part however, `\s${MAIL_ACCOUNT}` should be reliable, and then it should either end with `$` or `\s`~~ (**EDIT:** My bad, the config uses `,` to delimit multiple values for an alias).

---

`grep` pattern should match the account with these bounds: (`\s` || `,`) + `${MAIL_ACCOUNT}` + (`,` || `$`)

Reply:
Should be fixed with https://github.com/docker-mailserver/docker-mailserver/pull/2881/commits/2f4ed0b083a153de5275049c2d4d1e6136983ad1